### DISCUSSION TITLE
ArchiveDB

### DISCUSSION THREAD

**Discussion by mtgnoah** [ava-labs/hypersdk#233] [Chat]
Hey I would love to help contribute to the archive database for hypersdk chains. I was thinking of doing snapshot's similar to geth because they both us pebbledb but I wanted to ask what people thought before I started working on it.

**Comment by patrick-ogrady** [comment]
As mentioned in this issue, geth's snapshots are not used to serve historical data: https://github.com/ava-labs/avalanchego/issues/1715

We have some ideas for how this will look and it will likely be an "add on" for the MerkleDB. We'll use a trie-node less structure where you iterate over a key's changes to determine what value it had at a given height (somewhat similar to reth/erigon model).

**Comment by patrick-ogrady** [comment]
@nytzuga took a first pass at this: https://github.com/ava-labs/avalanchego/pull/1911

**Comment by patrick-ogrady** [comment]
This was completed and is now ready for use: https://github.com/ava-labs/avalanchego/tree/master/x/archivedb
